### ISSUE TITLE
Handling of PHP versions mismatch

### ISSUE THREAD

**Issue by stevector** [pantheon-systems/terminus-build-tools-plugin#92]
@ataylorme, @greg-1-anderson, I was working with an agency today and running the `build:project:create` command with WordPress. We were getting errors because The WordPress repo currently requires PHP 7.1. (example-wordpress-composer requires WordHat requires, which requires `sensiolabs/behat-page-object-extension` which requires `ocramius/proxy-manager` which require 7.1)

I don't think we can expect many people to have 7.1 locally. Should we include `--no-dev` on the composer create-project command that `build:project:create` wraps?

**Comment by greg-1-anderson** [comment]
Are there no older versions of those projects that works with php 7.0, at least?  WordHat [claims to require php 5.6](https://wordhat.info/getting-started/installation/index.html), so it seems there must be some way around this.

Maybe someone used php 7.1 to generate the composer.lock file? Did you try removing composer.lock and vendor, and running composer install again?  We should get in the habit of creating our lock files with php 5.6.

**Comment by ataylorme** [comment]
@stevector and @greg-1-anderson looking at the `composer.lock` for Example WordPress Composer there are 17 instances of packages requiring PHP 7 or greater and 3 instances of PHP 7.1.

Dropping to PHP 5.6. support for `build:project:create` would require us to be stuck on old versions of 17 dependencies, which seems risky.

I have an alternate proposal: Document the PHP `7.1` requirement but also setup a secure form where folks can enter the information needed by `build:project:create` and then we run the command on their behalf. 

This could be done on Circle CI or elsewhere we can have the PHP version be `7.1`. Having a webform would also require less setup in trainings. Rather than asking someone to install build tools for the one time project creation we can simply have them fill out the form.

**Comment by greg-1-anderson** [comment]
@ataylorme @stevector If you remove composer.lock and the vendor directory, and run `composer install` with php 5.6, then the composer.lock file should no longer require php 7.0 or 7.1.

**Comment by ataylorme** [comment]
@greg-1-anderson even if dependencies require PHP 7.0 or 7.1? Also if we remove `composer.lock` wouldn't we need to regenerate with `composer update` not `composer install`?

**Comment by greg-1-anderson** [comment]
@ataylorme WordHat says that it requires php 5.6, so `composer install` should work with that version. However, if you run `composer update` on a system that is running php 7.1, you might get a newer version of some of the dependencies than you would get if you ran the same operation using php 5.6.

You are correct that `composer install` and `composer update` do the same thing when there is no lock file.

**Comment by stevector** [comment]
I should have done more testing on my own before opening this issue. I switched back to PHP 5.6 on my own machine and I was able to get `example-wordpress-composer` to install just by deleting `composer.lock` and changing `"phpunit/phpunit": "^6.1"` to `"phpunit/phpunit": "^5.7 || ^6.1"`.

Yes, lots of packages that say they require PHP 7 get installed when running `composer install` with PHP 7 installed. But their older versions get used when switching back to PHP 5.

> We should get in the habit of creating our lock files with php 5.6.

Maybe. I'd at least like to get back down to 7.0 since that is the highest documented version on Pantheon. I also remembered that I raised this issue last month over in https://github.com/pantheon-systems/example-wordpress-composer/issues/8
